test/Microsoft.AspNetCore.SystemWebAdapters.Tests/HttpPostedFileTests.cs

```diff
@@ -100,4 +100,15 @@ public void InputStream()
         // Assert
         Assert.Equal(expected.Object, stream);
     }
+
+    [Fact]
+    public void SaveAs()
+    {
+        var file = new Mock<IFormFile>();
```

Comment:
can you follow the arrange/act/assert pattern as above?  Can you add a test for something rooted? maybe to the temp directory?

Reply:
@twsouthwick does this look ok ? (Dividing into 2 separate test cases)

```csharp
    [Fact]
    public void SaveAsForInvalidRootPath()
    {
        // Arrange
        var file = new Mock<IFormFile>();
        var expectedStream = new Mock<Stream>();
        file.Setup(f => f.OpenReadStream()).Returns(expectedStream.Object);
        var posted = new HttpPostedFile(file.Object);

        //Act and Assert
        Assert.Throws<HttpException>(() => posted.SaveAs("InvalidPath"));
    }

    [Fact]
    public void SaveAsWithValidRootPath()
    {
        // Arrange
        var file = new Mock<IFormFile>();
        var expectedStream = new Mock<Stream>();
        file.Setup(f => f.OpenReadStream()).Returns(expectedStream.Object);
        string validTempPath = string.Format(CultureInfo.InvariantCulture, @"{0}{1}.txt", Path.GetTempPath(), Guid.NewGuid());
        var posted = new HttpPostedFile(file.Object);

        //Act
        posted.SaveAs(validTempPath);

        //Assert
        Assert.True(File.Exists(validTempPath), "Temp file should be created");

        //Cleanup
        File.Delete(validTempPath);
    }
```